src/firmware/shared/circular_buffer.c

```diff
@@ -0,0 +1,74 @@
+#include "firmware/shared/circular_buffer.h"
+
+#include <assert.h>
+
+struct CircularBuffer
+{
+    int head;
+    int tail;
```

Comment:
I think these can be unsigned

Reply:
Changed to `uint8_t`

Reply:
you'll need to assert that `size` in `circular_buffer_create(size_t size)` is less than 256 and add a test for that

Reply:
What's the easiest way to unit test that in `SetUp`? Since when you run the test, it'll automatically call `SetUp` and can't assert it below.

Reply:
Again, why not define these as `size_t`? `uint_8` is a different size than `size_t`, so it's a bit weird having the `create` function take `size_t` for the size but then only being able to address values with `uint_8`

Reply:
Changed to use `size_t` for `size`, `head`, and `tail`.